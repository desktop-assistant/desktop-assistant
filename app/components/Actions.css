.topActions {
  z-index: 1;
  position: absolute;
  top: 0;
  right: 0;
  margin: 10px 5px;
}

.topActions > * {
  margin: 5px;
}

.bottomActions {
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 10px;
}

.sync {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 28px;
  height: 28px;
  border-radius: 50%;
  background-color: hsl(205, 70%, 58%);
  margin-bottom: 6px;
}

a.syncButton {
  font-size: 18px;
  color: white;
}

.expand, .settings {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 24px;
  height: 24px;
  border-radius: 50%;
	background-color: rgba(0, 0, 0, 0.6);
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
}

.expandButton, .settingsButton {
  font-size: 14px;
  color: white;
  border: none;
  background: none;
  padding: 0;
}

.settingsButton {
font-size: 10px;
}

.add {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 28px;
  height: 28px;
  border-radius: 50%;
	background-color: #b8e986;
	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
}

.add > a {
  display: flex;
  justify-content: center;
  align-items: center;
}

a.addTaskButton {
  font-size: 18px;
  color: #417505;
}
