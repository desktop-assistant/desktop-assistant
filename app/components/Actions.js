// @flow
import React, { Component } from 'react';
import { remote, screen } from 'electron';
import { Link } from 'react-router-dom';
import styles from './Actions.scss';

export default class Actions extends Component {
  state: {
    expanded: boolean
  };

  constructor() {
    super();
    this.state = {
      expanded: false
    };
  }

  previousPosition = []
  previousSize = []

  expandWindow() {
    const mainScreen = screen.getPrimaryDisplay();
    const win = remote.getCurrentWindow();

    if (this.state.expanded) {
      this.setState({ expanded: false });
      win.setPosition(this.previousPosition[0], this.previousPosition[1], true);
      win.setSize(this.previousSize[0], this.previousSize[1], true);
    } else {
      this.setState({ expanded: true });
      this.previousPosition = win.getPosition();
      this.previousSize = win.getSize();
      win.setPosition(this.previousPosition[0], 0, true);
      win.setSize(300, mainScreen.workArea.height, true);
    }
  }

  render() {
    return (
      <div>
        <div className={styles.topActions}>
          <button className={styles.settingsButton}>
            <Link to="/settings">
              <i className="fa fa-cog" />
            </Link>
          </button>
          <button className={styles.expandButton}>
            <a onClick={this.expandWindow.bind(this)}>
              <i className={`fa ${this.state.expanded ? 'fa-angle-double-up' : 'fa-angle-double-down'}`} />
            </a>
          </button>
        </div>
        <div className={styles.bottomActions}>
          <button className={styles.addTaskButton}>
            <Link to="/add">
              <i className="fa fa-plus" />
            </Link>
          </button>
        </div>
      </div>
    );
  }
}
