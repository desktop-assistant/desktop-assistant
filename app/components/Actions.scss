.topActions {
  z-index: 2;
  position: absolute;
  top: 0;
  right: 0;

	button, button a {
		color: white;
	}

	.expandButton {
		font-size: 14px;
	}
}

.topActions, .bottomActions {
  & > * {
    margin: 5px;
  }

	button {
		border: none;

		a {
			display: block;
			height: 100%;
		}
	}
}

.bottomActions {
  z-index: 2;
  position: absolute;
  bottom: 0;
  right: 0;
}

.sync {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 28px;
  height: 28px;
  border-radius: 50%;
  background-color: hsl(205, 70%, 58%);
  margin-bottom: 6px;
}

a.syncButton {
  font-size: 18px;
  color: white;
}

.expandButton, .settingsButton {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 24px;
  height: 24px;
  border-radius: 50%;
	background-color: rgba(0, 0, 0, 0.6);
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
}

.settingsButton {
	font-size: 10px;
}

.addTaskButton {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 28px;
  height: 28px;
  border-radius: 50%;
	background-color: #b8e986;
	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);

	a {
		font-size: 18px;
		color: #417505;
	}
}

.add > a {
  display: flex;
  justify-content: center;
  align-items: center;
}
