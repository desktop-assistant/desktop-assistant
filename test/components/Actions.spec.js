import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Actions from '../../app/components/Actions';

describe('Actions component', () => {
  it('should match exact snapshot', () => {
    const tree = renderer
      .create(
        <div>
          <Router>
            <Actions />
          </Router>
        </div>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
