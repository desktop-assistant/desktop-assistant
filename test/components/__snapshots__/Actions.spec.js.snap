// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions component should match exact snapshot 1`] = `
<div>
  <div>
    <div
      className="topActions"
    >
      <div
        className="settings"
      >
        <a
          className="settingsButton"
          href="/settings"
          onClick={[Function]}
        >
          <i
            className="fa fa-cog"
          />
        </a>
      </div>
      <div
        className="expand"
      >
        <button
          className="expandButton"
          onClick={[Function]}
        >
          <i
            className="fa fa-angle-double-down"
          />
        </button>
      </div>
    </div>
    <div
      className="bottomActions"
    >
      <div
        className="add"
      >
        <a
          className="addTaskButton"
          href="/add"
          onClick={[Function]}
        >
          <i
            className="fa fa-plus"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
